### PR TITLE
Automatically delete issues channel when sync is requested

### PR DESCRIPTION
**Pull request by DebitCardz**

### CODE REVIEW

File: src/commands/SyncCommand.ts [hunk 1]
```diff
@@ -65,6 +80,21 @@ export default class SyncCommand extends Command {
     );
   }
 
+  private async resetIssuesChannel(guild: Guild): Promise<GuildChannel> {
+    // Remove all previous issue channels so we can reinitialize it.
+    for (const matchingChannel of await guild.channels.array()) {
+      if (matchingChannel.name == config.discord.channels.issues.name) {
```
Comment (CoreyShupe) [inline src/commands/SyncCommand.ts:86]: triple equals here please

File: src/commands/SyncCommand.ts [hunk 2]
```diff
@@ -65,6 +80,21 @@ export default class SyncCommand extends Command {
     );
   }
 
+  private async resetIssuesChannel(guild: Guild): Promise<GuildChannel> {
+    // Remove all previous issue channels so we can reinitialize it.
+    for (const matchingChannel of await guild.channels.array()) {
```
Comment (CoreyShupe) [inline src/commands/SyncCommand.ts:85]: you might want to base array off parent here so we're only getting the channels under the parent, think an issues channel under category "a" and category "b", then this code breaks.

File: src/types/Config.ts [hunk 3]
```diff
@@ -3,7 +3,10 @@ export default interface Config {
     secret: string;
     channels: {
       logs: string;
-      issues: string;
+      issues: {
+        parent: string | undefined;
```
Comment (CoreyShupe) [inline src/types/Config.ts:7]: use `parent?: string` instead of `parent: string | undefined`

File: src/commands/SyncCommand.ts [hunk 4]
```diff
@@ -13,17 +21,24 @@ export default class SyncCommand extends Command {
   async execute(ctx: CommandContext) {
     const start = Date.now();
 
+    // DM channel or something?
+    if (ctx.guild == null) {
```
Comment (CoreyShupe) [inline src/commands/SyncCommand.ts:25]: triple equals here, and don't use null use undefined

**Review by CoreyShupe — CHANGES_REQUESTED** [review]

File: src/commands/SyncCommand.ts (hunk 2 above)
Comment (DebitCardz) [reply src/commands/SyncCommand.ts:85]: I don't quite get what you mean by "base array off parent here", you mean get channels under a specific parent?

Comment (CoreyShupe) [reply src/commands/SyncCommand.ts:85]: yes that's what I mean, right now you're getting all channels and just raw checking the name, if that name exists outside of the specified parent you kind of get screwed here

File: src/commands/SyncCommand.ts [hunk 5]
```diff
@@ -1,29 +1,46 @@
...
+import { trueEquals } from "../util/functions.ts";
 
 export default class SyncCommand extends Command {
   name = "sync";
 
   async execute(ctx: CommandContext) {
     const start = Date.now();
 
+    // DM channel or something?
+    if (ctx.guild === undefined) {
```
Comment (CoreyShupe) [inline src/commands/SyncCommand.ts:26]: does this return null as well? should we be using trueEquals here

File: src/commands/SyncCommand.ts [hunk 6]
```diff
@@ -1,29 +1,46 @@
...
+      return;
+    }
+
     const replyMessage = await ctx.message.reply(
       `Synchronizing issues with **${config.github.repo}**.`,
     );
 
+    // We can store this somewhere else later, for now this works.
```
Comment (CoreyShupe) [inline src/commands/SyncCommand.ts:34]: can you put a "TODO" market in this comment please

File: src/util/functions.ts [hunk 7]
```diff
@@ -0,0 +1,4 @@
+/**
```
Comment (CoreyShupe) [inline src/util/functions.ts:1]: if you're doing doc comments do them properly

**Review by CoreyShupe — CHANGES_REQUESTED** [review]

File: src/commands/SyncCommand.ts (hunk 5 above)
Comment (DebitCardz) [reply src/commands/SyncCommand.ts:26]: I actually don't know, it was just throwing an error saying it could be undefined.

Comment (CoreyShupe) [reply src/commands/SyncCommand.ts:26]: Instead of `=== undefined` just use `if (!ctx.guild)` then

Comment (DebitCardz) [reply src/commands/SyncCommand.ts:26]: Fair, will fix.

**Comment by CoreyShupe** [comment]
Looks good to me

**Pull request merged**